homeassistant/components/media_player/firetv.py

```diff
@@ -4,166 +4,144 @@
+    return cv.isfile(priv_key)
+
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Optional(CONF_DEVICE, default=DEFAULT_DEVICE): cv.string,
-    vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+    vol.Required(CONF_HOST): cv.string,
     vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
     vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-    vol.Optional(CONF_SSL, default=DEFAULT_SSL): cv.boolean,
+    vol.Optional(CONF_ADBKEY): has_adb_files,
+    vol.Optional(CONF_GET_SOURCE, default=DEFAULT_GET_SOURCE): cv.boolean,
+    vol.Optional(CONF_GET_SOURCES, default=DEFAULT_GET_SOURCES): cv.boolean
 })
 
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the FireTV platform."""
-    name = config.get(CONF_NAME)
-    ssl = config.get(CONF_SSL)
-    proto = 'https' if ssl else 'http'
-    host = config.get(CONF_HOST)
-    port = config.get(CONF_PORT)
-    device_id = config.get(CONF_DEVICE)
-
-    try:
-        response = requests.get(
-            DEVICE_LIST_URL.format(proto, host, port)).json()
-        if device_id in response[CONF_DEVICES].keys():
-            add_entities([FireTVDevice(proto, host, port, device_id, name)])
-            _LOGGER.info("Device %s accessible and ready for control",
-                         device_id)
-        else:
-            _LOGGER.warning("Device %s is not registered with firetv-server",
-                            device_id)
-    except requests.exceptions.RequestException:
-        _LOGGER.error("Could not connect to firetv-server at %s", host)
+PACKAGE_LAUNCHER = "com.amazon.tv.launcher"
+PACKAGE_SETTINGS = "com.amazon.tv.settings"
 
 
-class FireTV:
-    """The firetv-server client.
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the FireTV platform."""
+    from firetv import FireTV  # pylint: disable=no-name-in-module
+
+    host = '{0}:{1}'.format(config[CONF_HOST], config[CONF_PORT])
+
+    if CONF_ADBKEY in config:
+        ftv = FireTV(host, config[CONF_ADBKEY])
+        adb_log = " using adbkey='{0}'".format(config[CONF_ADBKEY])
+    else:
+        ftv = FireTV(host)
+        adb_log = ""
+
+    if not ftv.available:
+        _LOGGER.warning("Could not connect to Fire TV at %s%s", host, adb_log)
+    else:
+        name = config[CONF_NAME]
+        get_source = config[CONF_GET_SOURCE]
+        get_sources = config[CONF_GET_SOURCES]
+
+        device = FireTVDevice(ftv, name, get_source, get_sources)
+        add_devices([device])
+        _LOGGER.info("Setup Fire TV at %s%s", host, adb_log)
+
+
+def adb_decorator(override_available=False):
+    """Send an ADB command if the device is available and not locked."""
+    def adb_wrapper(func):
+        """Wait if previous ADB commands haven't finished."""
+        @functools.wraps(func)
+        def _adb_wrapper(self, *args, **kwargs):
+            # Only proceed if the device is available
+            if self.available or override_available:
```

Comment:
Invert this and `return None` if true.

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/17767/commits/856dee56a1382e8df946540e5ebecd3a41092426